test/controllers/FactCheckPageReviewControllerSpec.scala

```diff
@@ -0,0 +1,200 @@
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.http.{MimeTypes, Status}
+import play.api.mvc._
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import uk.gov.hmrc.http.HeaderCarrier
+import views.html.fact_check_page_review
+
+import scala.concurrent.Future
+
+class FactCheckPageReviewControllerSpec extends ControllerBaseSpec with GuiceOneAppPerSuite with MockReviewService {
+
+  private trait Test extends ReviewData {
+
+    implicit val hc: HeaderCarrier = HeaderCarrier()
+
+    val errorHandler: ErrorHandler = injector.instanceOf[ErrorHandler]
+    val view: fact_check_page_review = injector.instanceOf[fact_check_page_review]
+    val formProvider = new FactCheckPageReviewFormProvider()
+
+    val reviewController =
+      new FactCheckPageReviewController(errorHandler, FakeFactCheckerIdentifierAction, formProvider, view, mockReviewService, messagesControllerComponents)
+
+    val fakeRequest: FakeRequest[AnyContentAsEmpty.type] = FakeRequest("POST", "/")
+  }
+  "The fact check controller when loading the page" should {
+    "Return OK when the page data is retrieved successfully" in new Test {
+
+      MockReviewService
+        .factCheckPageInfo(id, reviewDetail.pageUrl)
+        .returns(Future.successful(Right(reviewDetail)))
+
+      val result: Future[Result] = reviewController.onPageLoad(id, updatedReviewDetail.pageUrl.drop(1))(fakeRequest)
+
+      status(result) shouldBe Status.OK
+    }
+
+    "Return OK when the page data is retrieved successfully and contains a result" in new Test {
+
+      MockReviewService
+        .factCheckPageInfo(id, reviewDetail.pageUrl)
+        .returns(Future.successful(Right(updatedReviewDetail)))
+
+      val result: Future[Result] = reviewController.onPageLoad(id, updatedReviewDetail.pageUrl.drop(1))(fakeRequest)
+
+      status(result) shouldBe Status.OK
+    }
+
+    "Return InternalServerError when the page data fails to retrieve" in new Test {
+
+      MockReviewService
+        .factCheckPageInfo(id, reviewDetail.pageUrl)
+        .returns(Future.successful(Left(StaleDataError)))
+
+      val result: Future[Result] = reviewController.onPageLoad(id, updatedReviewDetail.pageUrl.drop(1))(fakeRequest)
+
+      status(result) shouldBe Status.INTERNAL_SERVER_ERROR
+    }
+  }
+
+  "The fact check controller when submitting the page" should {
+
+    "Return SEE_OTHER for a successful post of the review completion for a process" in new Test {
+
+      MockReviewService
+        .factCheckPageComplete(id, updatedReviewDetail.pageUrl, updatedReviewDetail)
+        .returns(Future.successful(Right(())))
+
+      val result: Future[Result] = reviewController.onSubmit(id, updatedReviewDetail.pageUrl.drop(1))(fakeRequest.withFormUrlEncodedBody(("answer", "Yes")))
+
+      status(result) shouldBe Status.SEE_OTHER
+    }
+
+    "Return an Html document displaying the details of the review result" in new Test {
```

Comment:
Can it really be an HTML document if the content type is None?

Reply:
Odd, this a copy of the 2i page review tests and that passes also

Reply:
As the response is a redirect there is probably only a status code and location. As such there is no Html body. Should really change the description.